src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1027,76 +943,116 @@
        (when include-hash?
          {:hash (sutils/munge-hash-for-storage
                  (shash/fact-identity-hash fact-data))})))
-    ;; Ensure that all the required paths and values exist, and then
-    ;; insert the new facts.
     (let [paths-and-valuemaps (facts/facts->paths-and-valuemaps values)
           pathstrs (map (comp facts/factpath-to-string first) paths-and-valuemaps)
-          valuemaps (map second paths-and-valuemaps)
-          vhashes (map :value_hash valuemaps)
-          paths-to-ids (realize-paths! pathstrs)
-          vhashes-to-ids (realize-values! valuemaps)
-          pairs (map #(vector (get paths-to-ids %1)
-                              (get vhashes-to-ids %2))
-                     pathstrs vhashes)]
-      (insert-facts-pv-pairs! (certname-to-factset-id certname)
-                              pairs)))))
+          paths->ids (realize-paths! pathstrs)
+          valuemaps (map second paths-and-valuemaps)]
+      (insert-facts! (certname-to-factset-id certname)
+                     (map paths->ids pathstrs)
+                     valuemaps)))))
+
+(defn- update-existing-values [factset-id change-info]
+  (when (seq change-info)
+    (sql/db-do-prepared
+     jdbc/*db* true
+     (cons
+      (str "update facts set"
+           "    value_type_id = ?,"
+           "    value_boolean = ?,"
+           "    value_integer = ?,"
+           "    value_float = ?,"
+           "    value_string = ?,"
+           "    value = ?,"
+           "    large_value_hash = ?"
```

Comment:
this fails because it's using the wrong type for large_value_hash.

```
2017-02-03 10:21:44 PST ERROR:  column "large_value_hash" is of type bytea but expression is of type character varying
at character 167
```